app/forms/sections/documents/DocumentReferenceFormProvider.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.sections.documents
+
+import forms.XSS_REGEX
+import forms.mappings.Mappings
+import play.api.data.Form
+
+import javax.inject.Inject
+
+class DocumentReferenceFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[String] =
+    Form(
+      "value" -> text("documentReference.error.required")
+        .transform[String](
```

Comment:
This transformation looks really useful. 👍🏿 will it be worth having different `text()` e.g. `normalisedSpaceText()` which does this automatically?